File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int>set;
        for(int n:nums) set.insert(n);
        int len =0;
        for(int n:set){
            int curNum = n;
            if(set.contains(n+1)){
                continue;
            }
            int curLen =0;
            while(set.contains(curNum)){
                curLen++;
                curNum -= 1;
            }
            len = max(len, curLen);
        }
        return len;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int len =0;
        HashSet<Integer>set = new HashSet<>();
        for(int num:nums) set.add(num);
        for(int num:nums){
            if(set.contains(num+1)){
                continue;
            }
            int curNum = num;
            int lenCur=0;
            while(set.contains(curNum)){
                lenCur ++;
                set.remove(curNum);
                curNum -=1;
            }
            len = Math.max(len, lenCur);
        }
        return len;
    }
}